File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long total = 0;
        long long zeros = 0;

        for(auto i:nums){
            if(i == 0){
                zeros++;
            }else{
                zeros=0;
            }

            total += zeros;
        }

        return total;
    }
};

File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        total = 0
        zeros = 0

        for i in nums:
            if i == 0:
                zeros+=1
            else:
                zeros = 0
            
            total += zeros
        
        return total
